app.rb

```diff
@@ -0,0 +1,169 @@
+
+    case gets.chomp
+    when '1'
+      colorize_output(36, 'Enter Student details')
+      colorize_outprint(35, 'Name: ')
+      name = gets.chomp
+
+      colorize_outprint(35, 'Age: ')
+      age = gets.chomp
+
+      colorize_outprint(35, 'Has parent permission? [Y/N]: ')
+      permission = gets.chomp.downcase
+      case permission
+      when 'y' then parent_permission = true
+      when 'n' then parent_permission = false
+      end
+
+      student = Student.new(age: age, name: name, parent_permission: parent_permission)
+      @people.push(student)
+      colorize_output(34, 'Person Student Created successfully')
+    when '2'
+      colorize_output(36, 'Enter Teacher details')
+      colorize_outprint(35, 'Name: ')
+      name = gets.chomp
+
+      colorize_outprint(35, 'Age: ')
+      age = gets.chomp
+
+      colorize_outprint(35, 'Specialization: ')
+      specialization = gets.chomp
+      @people.push(Teacher.new(age: age, name: name, specialization: specialization))
+      colorize_output(34, 'Person Teacher Created successfully')
+    end
+    display_list
+    continue
+  end
+
+  def create_book
+    colorize_output(36, 'Enter Book details')
+    colorize_outprint(35, 'Title: ')
+    title = gets.chomp
+
+    colorize_outprint(35, 'Author: ')
+    author = gets.chomp
+
+    @books.push(Book.new(title, author))
+
+    colorize_output(36, 'You have successfully created your book')
+    display_list
+    continue
+  end
+
+  def list_all_books_with_numbers
+    if @books.empty?
+      colorize_output(34, 'Please insert books first!')
+      display_list
+      continue
+    else
+      @books.each_with_index do |book, index|
+        puts "#{index}) Title: #{book.title}, Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_all_person_with_numbers
+    if @people.empty?
+      colorize_output(34, 'Please insert people first!!')
+      display_list
+      continue
+    else
+      @people.each_with_index do |person, index|
+        puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  def create_rental()
+    colorize_output(36, 'Select a book from the following list by number')
+    list_all_books_with_numbers
+    book_option = gets.chomp.to_i
+
+    colorize_output(36, 'Select a person from the following list by number (not id)')
+    list_all_person_with_numbers
+    person_option = gets.chomp.to_i
+
+    colorize_outprint(36, 'Date: ')
+    date = gets.chomp
+    Rental.new(date, @books[book_option], @people[person_option])
+
+    colorize_output(36, 'Rental created successfully')
+    display_list
+    continue
+  end
+
+  def list_rentals_of_person_id()
+    print 'Person ID: '
+    id = gets.chomp.to_i
+
+    rentals = @rentals.filter { |rental| rental.person.id == id }
+    puts 'List of Rentals:'
+    rentals.each do |rental|
+      puts "Date: #{rental.date}, Book '#{rental.book.title}' by #{rental.book.author}"
+    end
```

Comment:
- [ ] As you can see from the image below when I try to list the rentals using an ID I get no response despite 'creating a rental'. If you look at line 147 you are trying to loop over the `@rentals` array defined earlier in the file, but the problem is that in the `create_rental` method, on line 134, you are not adding the created rentals in the first place. Kindly fix that by pushing the rental to the array.

  ```
  @rentals.push(Rental.new(date, @books[book_option], @people[person_option]))
  ```

  ![uurir](https://user-images.githubusercontent.com/36542986/225004892-ddb61e02-4523-469d-94ae-38aceb953650.jpg)